Gopkg.toml

```diff
@@ -18,16 +18,20 @@ required = ["k8s.io/code-generator/cmd/client-gen"]
 
 [[constraint]]
   name = "github.com/palantir/k8s-spark-scheduler-lib"
-  version = "0.2.1"
+  version = "0.2.4"
 
 [[constraint]]
-  version = "kubernetes-1.14.9"
+  version = "kubernetes-1.17.2"
```

Comment:
is it possible to keep this in sync with the k8s dependency version of the lib? which is `0.15.7`: https://github.com/palantir/k8s-spark-scheduler-lib/blob/master/go.mod#L22

Reply:
Wonder if it is better to bump lib dependency instead of downgrading here? I can try it out but it took some time finally arriving at this version that works with all dependencies.

Reply:
Also found the hard way that all "k8s.io/client-go", "k8s.io/apiextensions-apiserver" and "k8s.io/apimachinery" need to be exact same versions.

Reply:
yea also I am confused on why lib's dependency is shown as `v0.15.7` instead of what it seems to be pulling `v1.15.7`, will have a look.
Alright `1.17.0` means we will support min 1.15 clusters, should be ok

Reply:
this explains the weird k8s versioning: https://github.com/kubernetes/apimachinery/releases/v0.15.7